packages/shorebird_cli/lib/src/commands/release_new/releaser.dart

```diff
@@ -0,0 +1,72 @@
+import 'dart:io';
+
+import 'package:args/args.dart';
+import 'package:shorebird_cli/src/commands/release_new/release_type.dart';
+import 'package:shorebird_cli/src/shorebird_env.dart';
+import 'package:shorebird_code_push_client/shorebird_code_push_client.dart';
+import 'package:shorebird_code_push_protocol/shorebird_code_push_protocol.dart';
+
+/// {@template releaser}
+/// Executes platform-specific functionality to create a release.
+/// {@endtemplate}
+abstract class Releaser {
+  /// {@macro releaser}
+  Releaser({
+    required this.argResults,
+    required this.flavor,
+    required this.target,
+  });
+
+  /// The arguments passed to the command.
+  final ArgResults argResults;
+
+  /// The flavor of the release, if any.
+  final String? flavor;
+
+  /// The target script to run, if any.
+  final String? target;
+
+  /// Whether --release-version must be specified to patch. Currently only
+  /// required for add-to-app/hybrid releases (aar and ios-framework).
+  bool get requiresReleaseVersionArg => false;
+
+  /// The type of artifact we are creating a release for.
+  ReleaseType get releaseType;
+
+  /// The root directory of the current project.
+  Directory get projectRoot => shorebirdEnv.getShorebirdProjectRoot()!;
+
+  /// Asserts that the command can be run.
+  Future<void> assertPreconditions();
+
+  /// Asserts that the combination arguments passed to the command are valid.
+  Future<void> assertArgsAreValid();
+
+  /// Builds the release artifacts for the given platform. Returns the "primary"
+  /// artifact for the platform (e.g. the AAB for Android, the IPA for iOS).
+  Future<FileSystemEntity> buildReleaseArtifacts();
```

Comment:
Since this class is class `Releaser` is it necessary to specify `release` in these APIs or can this be `buildArtifacts`, `uploadArtifacts`, etc. since release is implied in the class name?

Reply:
Good point. Going to leave as is and update once we've merged this in to avoid merge conflicts as we merge in the other PRs in this chain.